File: src/main/java/com/telegram/folobot/mappers/FoloPidorMapper.java
package com.telegram.folobot.mappers;

import com.telegram.folobot.domain.FoloPidorEntity;
import com.telegram.folobot.dto.FoloPidorDto;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class FoloPidorMapper {
    private final FoloUserMapper foloUserMapper;
    public FoloPidorDto mapToFoloPidorDto(FoloPidorEntity entity){
        return new FoloPidorDto()
                .setId(entity.getId())
                .setFoloUser(foloUserMapper.mapToFoloUserDto(entity.getFoloUserEntity()))
                .setScore(entity.getScore());
    }

    public FoloPidorEntity mapToFoloPidorEntity(FoloPidorDto dto){
        return new FoloPidorEntity()
                .setId(dto.getId())
                .setFoloUserEntity(foloUserMapper.mapToFoloUserEntity(dto.getFoloUser()))
                .setScore(dto.getScore());
    }
}

File: src/main/kotlin/com/telegram/folobot/mappers/FoloPidorMapper.kt
package com.telegram.folobot.mappers

import com.telegram.folobot.domain.FoloPidorEntity
import com.telegram.folobot.dto.FoloPidorDto
import org.springframework.stereotype.Component

@Component
class FoloPidorMapper(private val foloUserMapper: FoloUserMapper) {

    fun mapToFoloPidorDto(entity: FoloPidorEntity): FoloPidorDto {
        return FoloPidorDto(
            entity.id,
            foloUserMapper.mapToFoloUserDto(entity.foloUserEntity),
            entity.score,
            entity.lastWinDate,
            entity.lastActiveDate
        )
    }

    fun mapToFoloPidorEntity(dto: FoloPidorDto): FoloPidorEntity {
        return FoloPidorEntity(
            dto.id,
            foloUserMapper.mapToFoloUserEntity(dto.foloUser),
            dto.score,
            dto.lastWinDate,
            dto.lastActiveDate
        )
    }
}